### PR TITLE
Feature 12851 display confidential on user fields when no rights

### LINKED ISSUE #12851: Display 'Confidential' instead of no user when rights to see personal/sensitive data inside/outside jurisdiction are missing [1]
### Problem Description

Ticket #6667 introduced the 'Contact information' form for reporting and responsible users.

When a user does not have rights to see sensitive/personal data inside/outside their jurisdiction currently what is displayed in the 'Contact information' form (for reporting user and responsible user) is:
image

As for instance, there is not entity without a reporting user, the information displayed is not correct.

### Proposed Change

When the rights to sensitive/personal information inside/outside jurisdiction are missing, display confidential in the 'Contact information' form.

### Added Value/Benefit

Consistency and accuracy in what concerns the data displayed.

### Acceptance Criteria

- [ ] display 'Confidential' in the 'Contact information' forms of reporting and/or responsible users when the user that is logged in does not have rights to see sensitive/personal information inside/outside their jurisdiction;
- [ ] apply the change to any place in the web and mobile app where the 'Contact information' form for reporting and responsible users can be opened from.

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #12851

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12951

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12951

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java [hunk 1]
```diff
@@ -50,6 +53,7 @@ protected Component initContent() {
 		}
 		userCombo.setNullSelectionAllowed(true);
 		userCombo.setValue(getValue());
+		parentPseudonymizedFlag = parentPseudonymizedSupplier.get();
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java:56]: getting the parent pseudonymization flag here is useless here

`parentPseudonymizedSupplier.get()` should be called only when needed, when sowing the details popup

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java [hunk 2]
```diff
@@ -67,6 +71,7 @@ protected Component initContent() {
 
 		addValueChangeListener(c -> {
 			userCombo.setValue(c.getProperty().getValue());
+			parentPseudonymizedFlag = parentPseudonymizedSupplier.get();
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java:74]: same as above

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java]: removed and moved in the pop-up window code

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java]: removed and moved in the pop-up window code

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12951

**Review by leventegal-she — APPROVED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java [hunk 3]
```diff
@@ -31,6 +32,8 @@ public class UserField extends CustomField<UserReferenceDto> {
 	private boolean readOnly;
 	private boolean enabled;
 	private List<UserReferenceDto> items = new ArrayList<>();
+	private boolean parentPseudonymizedFlag;
```
Comment (leventegal-she) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java:35]: this field is redundant, please remove it

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java (hunk 3 above)
Comment (sergiupacurariu) [reply sormas-ui/src/main/java/de/symeda/sormas/ui/utils/UserField.java]: removed

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12951

**Pull request merged**